### PR TITLE
basic work on #13

### PR DESCRIPTION
**Pull request by marcelcorso**
Create a histogram specifying buckets

`curl -H "Content-type: application/json" -X POST -d '{"type": "histogram", "name": "server_handling_seconds", "help": "time spent handling reqs", "method": "observe", "value": 15.8, "buckets": [1,3,6,10,15,20]}' http://127.0.0.1:9091/`

pasted_image_at_2018-11-25__2_57_pm

Now don't say anything.. we should get the default ones

`curl -H "Content-type: application/json" -X POST -d '{"type": "histogram", "name": "server_handling_seconds_no_buckets", "help": "time spent handling reqs 2", "method": "observe", "value": 15.8}' http://127.0.0.1:9091/`

image 13

👍

### CODE REVIEW

File: delta.go [hunk 1]
```diff
@@ -24,12 +24,13 @@ const (
...
-	Method string            `json:"method"`
-	Value  float64           `json:"value"`
-	Labels prometheus.Labels `json:"labels"`
+	Type    MetricType        `json:"type"`
+	Name    string            `json:"name"`
+	Help    string            `json:"help"`
+	Method  string            `json:"method"`
+	Value   float64           `json:"value"`
+	Labels  prometheus.Labels `json:"labels"`
+	Buckets *[]float64        `json:"buckets"`
```
Comment (kostkobv) [inline delta.go]: Is there specific reason why this one should be a pointer?

Comment (marcelcorso) [reply delta.go]: Yes! If not specified we will use the default buckets. We need to leave that decision to the user. 
`[]float64{}` (empty slice) could have that meaning but I think it's weirder

Comment (kostkobv) [reply delta.go]: What if user would also provide empty slice? Shouldn't we use default buckets then as well? Also isn't it weirder to expect pointer to a pointer (slice is a pointer in fact)?

Comment (marcelcorso) [reply delta.go]: I read the prometheus source code https://github.com/prometheus/client_golang/blob/ea348d7c20ce64c3bccdac0311e8621285d1278a/prometheus/histogram.go#L149

and 
https://github.com/prometheus/client_golang/blob/master/prometheus/histogram.go#L182

so yeah. I'll do that too to simplify. thanks!

**Comment by marcelcorso** [comment]
@kostkobv came to my desk and said he approved it

**Pull request merged**